src/System.Private.CoreLib/src/System/Environment.CoreCLR.cs

```diff
@@ -77,6 +74,13 @@ public static string[] GetCommandLineArgs()
                 GetCommandLineArgsNative();
         }
 
+        private static string[] s_commandLineArgs;
+
+        internal static void SetCommandLineArgs(string[] cmdLineArgs) // invoked from VM
+        {
+            s_commandLineArgs = cmdLineArgs;
+        }
```

Comment:
This will break corert.  You would need to make a similar change there.

Reply:
Should I if-def it out instead?

Reply:
I'd prefer to avoid adding a layer of ifdefs.  We can use partial files, or we can use ifdefs; using both for the same things just makes my head hurt.

I've been trying to manage this change across corefx, coreclr, and corert, with lots of moving pieces. Thankfully now corefx is out of the picture.  Of course mono enters the picture. Anything that's now moved between the shared and not shared portions needs to be coordinated.

Can I ask that you let this land in corert and then review how to move things around across shared/not-shared across the three repos?